src/app/server/Mdns.cpp

```diff
@@ -144,6 +151,76 @@ CHIP_ERROR AdvertiseCommisioning()
     }
     advertiseParameters.SetShortDiscriminator(static_cast<uint8_t>(value & 0xFF)).SetLongDiscrimininator(value);
 
+    if (DeviceLayer::ConfigurationMgr().GetDeviceType(value) != CHIP_NO_ERROR)
+    {
+        ChipLogProgress(Discovery, "DNS-SD Device Type not set");
+    }
+    else
+    {
+        advertiseParameters.SetDeviceType(chip::Optional<uint16_t>::Value(value));
+    }
+
+    char deviceName[129];
+    if (DeviceLayer::ConfigurationMgr().GetDeviceName(deviceName, sizeof(deviceName)) != CHIP_NO_ERROR)
+    {
+        ChipLogProgress(Discovery, "DNS-SD Device Name not set");
+    }
+    else
+    {
+        advertiseParameters.SetDeviceName(chip::Optional<const char *>::Value(deviceName));
+    }
+
+#if CHIP_ENABLE_ROTATING_DEVICE_ID
+    char rotatingDeviceIdHexBuffer[RotatingDeviceId::kHexMaxLength];
+    size_t rotatingDeviceIdHexBufferSize = 0;
+    ReturnErrorOnFailure(GenerateRotatingDeviceId(rotatingDeviceIdHexBuffer, rotatingDeviceIdHexBufferSize));
+    advertiseParameters.SetRotatingId(chip::Optional<const char *>::Value(rotatingDeviceIdHexBuffer));
```

Comment:
I believe this and the hints are the first instances of pointers inside optionals. We may want to specify the lifetime of these optionals to ensure they are used correctly. The pointers become invalid once this method ends.

Reply:
this object goes away after the Advertise call at the end of this method completes